deployment/ccip/changeset/cs_ccip_home.go

```diff
@@ -175,10 +286,154 @@ func SetCandidatePluginChangeset(
 		return deployment.ChangesetOutput{}, fmt.Errorf("%w: %w", deployment.ErrInvalidConfig, err)
 	}
 
+	txOpts := e.Chains[cfg.HomeChainSelector].DeployerKey
+	if cfg.MCMS != nil {
+		txOpts = deployment.SimTransactOpts()
+	}
+
 	newDONArgs, err := internal.BuildOCR3ConfigForCCIPHome(
 		e.OCRSecrets,
-		state.Chains[cfg.NewChainSelector].OffRamp,
-		e.Chains[cfg.NewChainSelector],
+		state.Chains[cfg.DONChainSelector].OffRamp,
+		e.Chains[cfg.DONChainSelector],
+		nodes.NonBootstraps(),
+		state.Chains[cfg.HomeChainSelector].RMNHome.Address(),
+		cfg.CCIPOCRParams.OCRParameters,
+		cfg.CCIPOCRParams.CommitOffChainConfig,
+		cfg.CCIPOCRParams.ExecuteOffChainConfig,
+	)
+	if err != nil {
+		return deployment.ChangesetOutput{}, err
+	}
+
+	latestDon, err := internal.LatestCCIPDON(state.Chains[cfg.HomeChainSelector].CapabilityRegistry)
+	if err != nil {
+		return deployment.ChangesetOutput{}, err
+	}
+
+	pluginOCR3Config, ok := newDONArgs[cfg.PluginType]
+	if !ok {
+		return deployment.ChangesetOutput{}, fmt.Errorf("missing commit plugin in ocr3Configs")
+	}
+
+	expectedDonID := latestDon.Id + 1
```

Comment:
> we'll need to find a way to prevent running a AddDonAndSetCandidateChangeset until the previous one is fully executed otherwise you can have 2 with the same expectedDonID

Yeah, had a similar thought, not sure of the best way to make this bullet proof.

Reply:
lets address in a follow up